### PR TITLE
Call Insight Reports API by name and date

### PR DESCRIPTION
**Pull request by owenbyrne**
Client can now call Insight Reports Endpoint and retrieve reports by name and date.

### CODE REVIEW

File: spec/insight_client_spec.rb [hunk 1]
```diff
@@ -155,4 +147,45 @@
...
+        it { expect(transaction).to eq [['col1', 'col2', 'col3'], [1, 2, 3], [4, 5, 6]] }
+      end
+
+      describe 'report does not exist' do
+        let(:expected_result) do
+          { data: [] }
+        end
+
+        let(:state)           { 'there is no report named daily_sales' }
+        let(:submission)      { 'a report request for daily_sales' }
```
Comment (DanielShuey) [inline spec/insight_client_spec.rb]: so much spaceage

File: spec/insight_client_spec.rb [hunk 2]
```diff
@@ -8,6 +8,14 @@
 describe InsightClient, pact: true do
   include_context 'pact consumer contract'
 
+  let(:response) do
+    {
+      headers: { 'Content-Type' => 'application/json' },
+      status: expected_status,
+      body: expected_result
+    }
```
Comment (DanielShuey) [inline spec/insight_client_spec.rb]: align :)

File: lib/insight-client.rb [hunk 3]
```diff
@@ -13,6 +13,10 @@ def retrieve(data, endpoint = :trips)
       transaction(:_get, data, endpoint)['trips'] || []
     end
 
+    def reports(data, endpoint = :reports)
+      transaction(:_get, data, endpoint)['data'] || []
+    end
```
Comment (DanielShuey) [inline lib/insight-client.rb]: Can we refactor this so we don't have 3 methods with default endpoints now. Happy to pair on it if you'd like.

File: spec/insight_client_spec.rb [hunk 4]
```diff
@@ -155,4 +147,45 @@
...
     end
   end
+
+   describe 'reports' do
+    let(:request) do
+      {
+        headers: { 'Content-Type' => 'application/json' },
+        method: :get,
+        path: '/api/reports',
+        query: query
```
Comment (DanielShuey) [inline spec/insight_client_spec.rb]: align :)

**Comment by owenbyrne** [comment]
@DanielShuey 

https://s-media-cache-ak0.pinimg.com/736x/2d/0d/50/2d0d50e0f7d9d1dcaf39a49d58db25ae.jpg

**Comment by owenbyrne** [comment]
Don't merge this yet - gonna tweak the tests a bit.

**Pull request merged**